src/shadowbox/server/manager_service.ts

```diff
@@ -168,6 +172,47 @@ export class ShadowsocksManagerService {
     }
   }
 
+  public async setAccessKeyQuota(req: RequestType, res: ResponseType, next: restify.Next) {
+    try {
+      logging.debug(`setAccessKeyQuota request ${JSON.stringify(req.params)}`);
+      const accessKeyId = req.params.id;
+      const quota = req.params.quota;
+      // TODO(alalama): remove these checks once the repository supports typed errors.
+      if (!quota || !quota.data || !quota.window) {
+        return next(new restify.InvalidArgumentError(
```

Comment:
I just realized that `restify.InvalidArgumentError` is returning HTTP error 409 Conflict.
That doesn't seem like a standard way of doing that.
Could you change to BadRequest (400) instead?